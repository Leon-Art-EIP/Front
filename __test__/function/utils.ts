import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
