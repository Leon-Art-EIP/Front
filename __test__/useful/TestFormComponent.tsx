import { FormProvider } from "react-hook-form";
import type { ReactNode } from "react";
import useTestForm from "./useTestForm";

interface ITestInputProps {
  name: string;
  value?: string[] | string;
  children: ReactNode;
}

export default function TestFormComponent(props: ITestInputProps): JSX.Element {
  const methods = useTestForm({ [props.name]: props.value });

  return <FormProvider {...methods}>{props.children}</FormProvider>;
}
