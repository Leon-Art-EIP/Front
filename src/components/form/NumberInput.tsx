"use client";

import { useController, useFormContext } from "react-hook-form";
import { cn } from "../../tools/cn";

interface IInputProps {
  name: string;
  className: string;
  placeholder?: string;
  title?: string;
}

export default function NumberInput(props: Omit<IInputProps, "type">): JSX.Element {
  const { register } = useFormContext();

  const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name: props.name });

  const handleInputOnChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    onChange(e.target.value);
  };

  return (
    <div className="flex flex-col gap-2 sm:min-w-min min-w-0">
      {props.title && <div className="text-lg text-tertiary">{props.title}</div>}
      <input
        type="number"
        name={props.name}
        id={props.name}
        className={cn("sm:min-w-min min-w-0", props.className)}
        placeholder={props.placeholder}
        onChange={handleInputOnChange}
        value={value}
      />
      {error && <div className="text-primary">{error.message}</div>}
    </div>
  );
}
