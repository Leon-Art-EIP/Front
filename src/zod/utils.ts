export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
