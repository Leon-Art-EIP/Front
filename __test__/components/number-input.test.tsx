import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TestFormComponent from "../useful/TestFormComponent";
import NumberInput from "../../src/components/form/NumberInput";

test("NumberInput snapshot", () => {
  testSnapshot(
    <TestFormComponent name="number">
      <NumberInput name="number" className="bg-white" />
    </TestFormComponent>
  );
});
