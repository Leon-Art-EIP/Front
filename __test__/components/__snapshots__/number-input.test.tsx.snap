// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NumberInput snapshot 1`] = `
<div
  className="flex flex-col gap-2 sm:min-w-min min-w-0"
>
  <input
    className="sm:min-w-min min-w-0 bg-white"
    id="number"
    name="number"
    onChange={[Function]}
    type="number"
  />
</div>
`;
